examples/03_Batching/inference-batcher.cc

```diff
@@ -0,0 +1,398 @@
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <chrono>
+#include <thread>
+
+#include "YAIS/YAIS.h"
+#include "moodycamel/blockingconcurrentqueue.h"
+
+using yais::Context;
+using yais::Executor;
+using yais::Server;
+using yais::ThreadPool;
+
+using moodycamel::BlockingConcurrentQueue;
+using moodycamel::ConsumerToken;
+using moodycamel::ProducerToken;
+
+#include "simple.pb.h"
+#include "simple.grpc.pb.h"
+
+
+/**
+ * @brief Batching Service for Unary Requests
+ * 
+ * Exposes a Unary (send/recv) interface for a given RPC, but rather than
+ * computing the RPC, the service simply batches the incoming requests and
+ * forwards them via a gRPC stream to a service that implements the actual
+ * compute portion of the RPC.
+ * 
+ * The backend compute service is not a Unary service.  Rather it must
+ * implemented the LifeCycleBatching service Context, i.e. BatchingContext.
+ * The other application in this folder implements the backend service.
+ * 
+ * Streams are used as a forwarding mechanism because of how they interact
+ * with a load-balancer.  Unlike unary requests which get balanced on each
+ * request, a stream only get balanced when it is opened.  All items of a stream
+ * go to the same endpoint service.
+ * 
+ * 
+ * 
+ * @tparam ServiceType 
+ * @tparam Request 
+ * @tparam Response 
+ */
+template <class ServiceType, class Request, class Response>
+struct BatchingService
+{
+    using Callback = std::function<void(bool)>;
+
+    struct MessageType
+    {
+        Request *request;
+        Response *response;
+        Callback callback;
+    };
+
+    /**
+     * @brief Forwards incoming Unary requests via a gRPC Stream to
+     * a Batched Steaming Service that implements the actual RPC
+     */
+    class Client
+    {
+      public:
+        Client(
+            std::shared_ptr<::grpc::Channel> channel,
+            std::shared_ptr<ThreadPool> thread_pool)
+            : m_Stub(ServiceType::NewStub(channel)), m_ThreadPool(thread_pool), m_CurrentCQ(0)
+        {
+            for (int i = 0; i < m_ThreadPool->Size(); i++)
```

Comment:
What is the data type of `m_ThreadPool->Size()`? If `size_t`, then `int i` will get the compiler to raise a warning.

Perhaps declare it as:
```cpp
for (size_t i = 0; i < m_ThreadPool->Size(); i++)
```

Reply:
I'll change it to `auto` or be specific.   

I was just being lazy.